### PR TITLE
Settings: Labeling for "Modified in ..." Fixes #59492

### LINKED ISSUE #59492: Settings: Update ARIA labeling to use 'Modified in (workspace /user settings)
Make labeling match element text

### PR DESCRIPTION
**Pull request by cleidigh**
- Modify main element labels to handle different "Modified" scopes.
- Tweak for checkbox oddities
- Use direct label on input elements - voiceover work around
- Add setElementAriaLabels to reduce duplicate code
- general labeling cleanup

Notes: not yet tested with folder scope changes
Tested with NVDA and OS X voiceover

@roblourens 
I moved some things around besides the cleanup.  The code checks and builds up modifiers was moved before renderElement so it could have access the element modifiers.  I don't think this has a problem, but just want to check with you before merging.

### CODE REVIEW

File: src/vs/workbench/parts/preferences/browser/settingsTree.ts [hunk 1]
```diff
@@ -1062,14 +1061,11 @@ export class SettingsRenderer implements ITreeRenderer {
 			template.descriptionElement.innerText = element.description;
 		}
 
+		// List row label overrides labeling - is not full label must use local label
+		template.containerElement.parentElement.removeAttribute('aria-label');
```
Comment (roblourens) [inline src/vs/workbench/parts/preferences/browser/settingsTree.ts]: I don't understand the comment. Is this removing the aria label from SettingsAccessibilityProvider?

File: src/vs/workbench/parts/preferences/browser/settingsTree.ts [hunk 2]
```diff
@@ -1293,6 +1229,59 @@ export class SettingsRenderer implements ITreeRenderer {
...
 
+
+	private setElementAriaLabels(dataElement: SettingsTreeSettingElement, templateId: string, template: ISettingItemTemplate): string {
+		// Create base Id for element references
+		const baseId = (dataElement.displayCategory + '_' + dataElement.displayLabel).replace(/ /g, '_').toLowerCase();
+
+		const modifiedText = template.otherOverridesElement.textContent ?
+			template.otherOverridesElement.textContent : (dataElement.isConfigured ? localize('settings.Modified', ' Modified. ') : '');
+
+		let itemElement = Object(HTMLElement);
```
Comment (roblourens) [inline src/vs/workbench/parts/preferences/browser/settingsTree.ts]: What is `Object(HTMLElement)`?

File: src/vs/workbench/parts/preferences/browser/settingsTree.ts (hunk 1 above)
Comment (cleidigh) [reply src/vs/workbench/parts/preferences/browser/settingsTree.ts]: It removes the label from the tree  row.  This label interferes with the local label the input element.
With voiceover it causes a duplicate label output with checkbox elements.

I am going to work on the general issue of role types and labeling variations for both tree and list.
Once I get that in we could do more generalization for setting and controlling these attributes as opposed to hack overrides.

File: src/vs/workbench/parts/preferences/browser/settingsTree.ts (hunk 2 above)
Comment (cleidigh) [reply src/vs/workbench/parts/preferences/browser/settingsTree.ts]: A stupid mistake, probably had Python or some other language of the brain.

Will change to <HTMLElement> Object.create(null)

Comment (roblourens) [reply src/vs/workbench/parts/preferences/browser/settingsTree.ts]: Why create a fake HTMLElement? I don't follow

Comment (cleidigh) [reply src/vs/workbench/parts/preferences/browser/settingsTree.ts]: you don't follow because it's hard to follow stupid things...

I don't need the object, I just need a placeholder variable since the type assertion later
takes care of the rest.  Too many languages doesn't mix with late-night programming.

File: src/vs/workbench/parts/preferences/browser/settingsTree.ts [hunk 3]
```diff
@@ -1293,6 +1229,59 @@ export class SettingsRenderer implements ITreeRenderer {
...
+			}
+		} else {
+			// Don't change attributes if we don't know what we areFunctions
+			return '';
+		}
+
+		// Labels will not be read on descendent input elements of the parent treeitem
+		// unless defined as roles for input items
+		// voiceover does not seem to use labeledby correctly, set labels directly on input elements
+		itemElement.id = baseId + '_setting_item';
```
Comment (roblourens) [inline src/vs/workbench/parts/preferences/browser/settingsTree.ts]: Will itemElement always be non-null here? Looks like there are null checks above like `if (itemElement = template.controlElement.firstElementChild)` but if that fails then it will still get to here

File: src/vs/workbench/parts/preferences/browser/settingsTree.ts (hunk 1 above)
Comment (cleidigh) [reply src/vs/workbench/parts/preferences/browser/settingsTree.ts]: I should do this within SettingsAccessibilityProvider if checkbox.

File: src/vs/workbench/parts/preferences/browser/settingsTree.ts (hunk 3 above)
Comment (cleidigh) [reply src/vs/workbench/parts/preferences/browser/settingsTree.ts]: sigh, yes. 👎

**Comment by cleidigh** [comment]
Setting the checkbox label to '' within SettingsAccessibilityProvider works with both
NVDA and voiceover

**Comment by cleidigh** [comment]
@roblourens 
Good with the syntax?

**Review by roblourens — APPROVED** [review]
LGTM
